### PR TITLE
[jit] Set debug name for value coming out of GetAttr nodes.

### PR DESCRIPTION
**Pull request by zhxchen17**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59124 [jit] Decouple unique id and debug name for displaying Value name.
* **#59123 [jit] Set debug name for value coming out of GetAttr nodes.**

Summary:
In https://github.com/pytorch/pytorch/issues/37387 we found that it could be
hard to debug the model without debug names attached to values out of
prim::GetAttr nodes. Therefore in this diff we set a debug name for such
node.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28766023](https://our.internmc.facebook.com/intern/diff/D28766023)

### CODE REVIEW

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28766023).

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28766023).

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28766023).

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28766023).

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28766023).

File: torch/csrc/jit/ir/ir.cpp [hunk 1]
```diff
@@ -24,23 +24,38 @@
...
+void printValueRef(std::ostream& out, const Value* n) {
   out << "%" << n->debugName();
 }
 
+bool isNumber(c10::string_view str) {
+  return str.find_first_not_of("0123456789") == std::string::npos;
+}
+
+std::string normalizeAttrName(c10::string_view field) {
+  if (isNumber(field)) {
```
Comment (SplitInfinity) [inline torch/csrc/jit/ir/ir.cpp:51]: Attribute names that are numbers are usually illegal in Python unless they are manually inserted into the object's instance dictionary. Why do you need to check whether `field` is a number?

Comment (SplitInfinity) [reply torch/csrc/jit/ir/ir.cpp:51]: I see that there are some existing tests that manage to use attributes starting with numbers, but I am wondering how...

Comment (zhxchen17) [reply torch/csrc/jit/ir/ir.cpp:51]: @SplitInfinity 
I'm seeing "0" used as attr name in ads unit tests... https://www.internalfb.com/intern/testinfra/diagnostics/5910974564491719.281474990955546.1622163900/

After some investigation, I found there's indeed one way:
```
class M(nn.Module):
     def __init__(self):
           setattr(self, "0", 0)
     def forward(self):
           return getattr(self, "0")
```
this way we got something like:
```
graph(%self : __torch__.M):
  %_0 : int = prim::GetAttr[name="0"](%self)
  return (%9)
```
and as the test result shows, this happens in real model...

Comment (SplitInfinity) [reply torch/csrc/jit/ir/ir.cpp:51]: Lol

**Review by SplitInfinity — APPROVED** [review]

**Pull request closed without merging**